app/models/shopify/import/base.rb

```diff
@@ -0,0 +1,69 @@
+module Shopify
+  module Import
+    class Base
+      def initialize(credentials: {}, client: nil)
+        @client = client(credentials) unless client
+      end
+
+      def count(**opts)
+        api_class.get(:count, opts)
+      end
+
+      def find(id)
+        api_class.find(id)
+      end
+
+      def find_and_import(id)
+        resource = find(id)
+        create_data_feed_and_spree_object(resource)
+      end
+
+      def find_all(**opts)
+        results = []
+        find_in_batches(**opts) do |batch|
+          return results if batch.nil?
+          batch.each { |resource| results << resource }
+        end
+        results
+      end
+
+      def find_all_first_and_import(**opts)
```

Comment:
This method name is confusing to me. I couldn't guess, without looking at the code, what it does. "Find all"...first products?
`find_and_import_all` or `find_all_and_import` or, use the `opts` param to pass the id(s) you want to find and "merge" `find_and_import` with this method. I think this is the best solution:

```ruby
def import(options = {})
  id = options.delete('id')
  if id.present?
    resource = find(id)
    create_data_feed_and_spree_object(resource)
  else
    find_all(**options).each do |resource|
      create_data_feed_and_spree_object(resource) # i'd call this process(resource) or transform(resource)
    end
  end
end
```